### PR TITLE
OEL-961: Fix dropbutton links.

### PR DESCRIPTION
**Pull request by escuriola**
Jira issue(s):
https://citnet.tech.ec.europa.eu/CITnet/jira/browse/OEL-961

### CODE REVIEW

File: includes/menu.inc [hunk 1]
```diff
@@ -78,31 +78,44 @@ function oe_bootstrap_theme_preprocess_links__dropbutton(&$variables) {
...
+  dropbutton_add_required_classes($links);
+}
 
+/**
+ * Add required classes to dropbutton links.
+ *
+ * @param array $links
+ *   Array with links items.
+ */
+function dropbutton_add_required_classes(array &$links): void {
```
Comment (drishu) [inline includes/menu.inc]:
```suggestion
function oe_bootstrap_theme_add_dropbutton_classes(array &$links): void {
```

File: includes/menu.inc [hunk 2]
```diff
@@ -78,31 +78,44 @@ function oe_bootstrap_theme_preprocess_links__dropbutton(&$variables) {
...
 
-  // Add required classes.
-  foreach ($links as $key => $link) {
-    $links[$key]['link']['#options']['attributes']['class'][] = 'dropdown-item';
+  $variables['button'] = $button;
+  dropbutton_add_required_classes($links);
+}
 
+/**
+ * Add required classes to dropbutton links.
```
Comment (drishu) [inline includes/menu.inc]:
```suggestion
 * Adds required classes to dropbutton links.
```

File: includes/menu.inc [hunk 3]
```diff
@@ -78,31 +78,44 @@ function oe_bootstrap_theme_preprocess_links__dropbutton(&$variables) {
...
-
     // Remove first link from links.
     array_shift($links);
   }
 
-  // Add required classes.
-  foreach ($links as $key => $link) {
-    $links[$key]['link']['#options']['attributes']['class'][] = 'dropdown-item';
+  $variables['button'] = $button;
+  dropbutton_add_required_classes($links);
```
Comment (drishu) [inline includes/menu.inc]:
```suggestion
  oe_bootstrap_theme_add_dropbutton_classes($links);
```

**Review by drishu — CHANGES_REQUESTED** [review]

**Review by drishu — DISMISSED** [review]

File: tests/src/Kernel/DropbuttonTest.php [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    $button = $crawler->filter('button.btn.btn-sm.btn-outline-primary');
+    $this->assertCount(1, $button);
+
+    $list = $crawler->filter('ul.dropdown-menu');
+    $this->assertCount(1, $list);
+
+    $items = $crawler->filter('li > a.dropdown-item');
+    $this->assertCount(5, $items);
+
+    $active = $crawler->filter('li:nth-child(1) a')->text();
```
Comment (abel-santos-corral) [inline tests/src/Kernel/DropbuttonTest.php]:
```suggestion
    $this->assertEquals('Link 1', trim($items->eq(0)->text()));
    $this->assertEquals('Link 2', trim($items->eq(1)->text()));
    $this->assertEquals('Link 3', trim($items->eq(2)->text()));
    $this->assertEquals('Link 4', trim($items->eq(3)->text()));
    $this->assertEquals('Link 5', trim($items->eq(4)->text()));
```

Comment (abel-santos-corral) [reply tests/src/Kernel/DropbuttonTest.php]: That can be done with fewer lines as the suggestion (already tested in local).

**Review by abel-santos-corral — APPROVED** [review]

**Review by drishu — APPROVED** [review]

**Pull request merged**